### PR TITLE
Created E2E tests for BasketController

### LINKED ISSUE #106: Create E2E tests for BasketController

### PR DESCRIPTION
**Pull request by MateuszHocholowski**

### CODE REVIEW

File: src/test/java/com/orzechazo/eshop/controllers/BasketControllerE2ETest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class BasketControllerE2ETest {
+
+    @Autowired
+    BasketRepository basketRepository;
```
Comment (RadekKaniecki) [inline src/test/java/com/orzechazo/eshop/controllers/BasketControllerE2ETest.java:36]: private

File: src/test/java/com/orzechazo/eshop/controllers/BasketControllerE2ETest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        DB_DEFAULT_BASKET_COUNT = bootstrap.getBaskets().size();
+    }
+
+    @Test
+    void getBasketByBasketId() throws Exception {
+        mockMvc.perform(get("/baskets/" + DB_BASKET_ID)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.basketId",equalTo(DB_BASKET_ID)))
+                .andExpect(jsonPath("$.totalPrice",equalTo(new BigDecimal("120").intValue())));
```
Comment (RadekKaniecki) [inline src/test/java/com/orzechazo/eshop/controllers/BasketControllerE2ETest.java:60]: constant

File: src/test/java/com/orzechazo/eshop/controllers/BasketControllerE2ETest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    @Test
+    void getBasketByBasketIdThatIsNotInDB() throws Exception {
+        mockMvc.perform(get("/baskets/basketNotInDb")
```
Comment (RadekKaniecki) [inline src/test/java/com/orzechazo/eshop/controllers/BasketControllerE2ETest.java:65]: basketNotInDb to constant

**Comment by MateuszHocholowski** [comment]
Corrected

File: src/test/java/com/orzechazo/eshop/bootstrap/tests/BootstrapBasket.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    public BootstrapBasket(BasketRepository basketRepository) {
+        this.basketRepository = basketRepository;
+    }
+    public void loadData() {
+        log.debug("loading Baskets...");
+
+        Basket basket1 = new Basket();
+        basket1.setBasketId(DB_BASKET_ID);
+        basket1.setTotalPrice(new BigDecimal("120"));
```
Comment (RadekKaniecki) [inline src/test/java/com/orzechazo/eshop/bootstrap/tests/BootstrapBasket.java:26]: BASKET_TOTAL_PRICE constant here :P

**Review by RadekKaniecki — APPROVED** [review]

**Pull request merged**